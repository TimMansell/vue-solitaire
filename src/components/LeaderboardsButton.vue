<template>
  <Button
    @click="$router.push('/leaderboards')"
    :is-stacked="true"
    :disabled="emptyBoard || !isOnline"
    data-test="leaderboards-btn"
  >
    Ranks
  </Button>
</template>

<script>
import { mapGetters } from 'vuex';
import Button from './Button.vue';

export default {
  name: 'LeaderboardsButton',
  components: {
    Button,
  },
  computed: {
    ...mapGetters(['emptyBoard', 'isOnline']),
  },
};
</script>
