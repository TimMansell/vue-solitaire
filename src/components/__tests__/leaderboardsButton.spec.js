import { shallowMount } from '@vue/test-utils';
import LeaderboardsButton from '@/components/LeaderboardsButton.vue';

describe('LeaderboardsButton.vue', () => {
  it('matches snapshot', () => {
    const wrapper = shallowMount(LeaderboardsButton);

    expect(wrapper).toMatchSnapshot();
  });
});
