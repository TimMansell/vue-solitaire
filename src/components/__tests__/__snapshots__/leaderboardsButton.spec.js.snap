// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeaderboardsButton.vue matches snapshot 1`] = `
<button-stub type="default" isstacked="true" checkdisabled="true" size="md" click="[Function]" route="leaderboards" data-test="leaderboards-btn">
  Ranks
</button-stub>
`;
